### PR TITLE
Some initial work on using words

### PR DESCRIPTION
**Pull request by ajreynol**
Towards support for sequences (CVC4/cvc4-projects#23.)

### CODE REVIEW

File: src/theory/strings/word.h [hunk 1]
```diff
@@ -29,50 +30,53 @@ namespace strings {
 class Word
 {
 public:
-  /** make empty constant of kind k */
-  static Node mkEmptyWord(Kind k);
+ /** make empty constant of type tn */
```
Comment (4tXJ7f) [inline src/theory/strings/word.h]: It looks like the formatting somehow got messed up (`public` is not indented at all and the `static` method only by one space. Can you ClangFormat the whole file?

File: src/theory/strings/theory_strings_rewriter.cpp [hunk 2]
```diff
@@ -2042,13 +2052,11 @@ Node TheoryStringsRewriter::rewriteContains( Node node ) {
...
     {
-      CVC4::String t = node[1].getConst<String>();
-      Node ret =
-          NodeManager::currentNM()->mkConst(s.find(t) != std::string::npos);
+      Node ret = nm->mkConst(Word::find(node[0], node[1]) != std::string::npos);
       return returnRewrite(node, ret, "ctn-const");
     }else{
       Node t = node[1];
-      if (s.size() == 0)
+      if (Word::getLength(node[0]) == 0)
```
Comment (4tXJ7f) [inline src/theory/strings/theory_strings_rewriter.cpp]:
```suggestion
      if (Word::isEmpty(node[0]))
```

File: src/theory/strings/theory_strings_rewriter.cpp [hunk 3]
```diff
@@ -970,8 +980,7 @@ Node TheoryStringsRewriter::rewriteStarRegExp(TNode node)
     return returnRewrite(node, node[0], "re-star-nested-star");
   }
   else if (node[0].getKind() == STRING_TO_REGEXP
-           && node[0][0].getKind() == CONST_STRING
-           && node[0][0].getConst<String>().isEmptyString())
+           && node[0][0].getKind() == CONST_STRING && Word::isEmpty(node[0][0]))
```
Comment (4tXJ7f) [inline src/theory/strings/theory_strings_rewriter.cpp]: We should have a method `Word::isWord(Node n)` that returns true for string/sequence constants and one for string/sequence regexps.

Comment (ajreynol) [reply src/theory/strings/theory_strings_rewriter.cpp]: Actually, we already do: `.isConst()` :)  That is the preferred way of checking this, indeed I should fix the kind comparison here.

Comment (ajreynol) [reply src/theory/strings/theory_strings_rewriter.cpp]: Btw, we will eventually need utility function for `isStringLike( TypeNode tn )` or something like that.

**Pull request merged**